src/controllers/MLFacetQueryController.ts

```diff
@@ -0,0 +1,34 @@
+import { MLFacet } from '../ui/MLFacet/MLFacet';
+import { QueryBuilder } from '../ui/Base/QueryBuilder';
+import { Assert } from '../misc/Assert';
+import { IFacetRequest, IFacetRequestValue } from '../rest/Facet/FacetRequest';
+import { FacetSortCriteria } from '../rest/Facet/FacetSortCriteria';
+
+export class MLFacetQueryController {
+  constructor(public facet: MLFacet) {}
+
+  /**
+   * Build the facets request for the MLFacet, and insert it in the query builder
+   * @param queryBuilder
+   */
+  public putFacetIntoQueryBuilder(queryBuilder: QueryBuilder) {
+    Assert.exists(queryBuilder);
+
+    const facetState: IFacetRequest = {
+      field: this.facet.fieldName,
```

Comment:
FieldName being a `get` method that transform the option by removing the "@", I have to make sure, is this actually needed ?

The API should support both sending a FacetRequest where the field name contains a `@`, and where the field name does not contain `@`.

This is the standard behaviour of other query parameters where we send a field name.